src/foam/u2/view/ChoiceView.js

```diff
@@ -269,9 +269,13 @@ foam.CLASS({
       isFramed: true,
       code: function() {
         var d = this.data;
-        if ( this.choices.length ) {
-          this.choice = ( d != null && this.findChoiceByData(d) ) || this.defaultValue;
+        var newChoice = this.choices.length && d != null &&
+          this.findChoiceByData(d);
+        if ( newChoice == this.choice ) {
```

Comment:
Wouldn't just removing the equality check in `choice.postSet` be sufficient to fix the bug?

I'm assuming we're only doing this check and setting it to null on the line below to make sure the postSet on the `choice` property gets triggered. Wouldn't just doing `this.choice = newChoice` work?

Reply:
if not adding this check, the data seem not to show up in the detail view after we selected and saved.